### ISSUE TITLE
Latest binary version (4.2.0) doesn't work in MacOS

### ISSUE THREAD

**Issue by nogates** [stoplightio/spectral#718]
**Describe the bug**

Latest binary version (4.20) doesn't work in MacOS 10.14.6 

**To Reproduce**

```
spectral ~/Downloads/spectral-macos lint openapi.yml
Error: @stoplight/path: Cannot parse /Users/nogates/Downloads/spectral-macos/openapi.yml because it is not a string
    at parse (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:6984)
    at Object.isAbsolute (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:7876)
    at Object.<anonymous> (/snapshot/spectral/dist/cli/services/linter.js:40:65)
    at Generator.next (<anonymous>)
    at /snapshot/spectral/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/snapshot/spectral/node_modules/tslib/tslib.js:106:16)
    at Object.lint (/snapshot/spectral/dist/cli/services/linter.js:36:20)
    at Object.handler (/snapshot/spectral/dist/cli/commands/lint.js:86:25)
    at Object.runCommand (/snapshot/spectral/node_modules/yargs/lib/command.js:235:26)
```
Also, version is not reported correctly

```
spectral ~/Downloads/spectral-macos --version
0.0.0
```

Previous version (4.1.1) works just fine

Tried both, using the script to pull the latest binary and manually downloaded it via GH releases

**Comment by philsturgeon** [comment]
I am a little confused by the command you are running here:

```
$ spectral ~/Downloads/spectral-macos lint openapi.yml
```

What happens if you run this:

```
$ ~/Downloads/spectral-macos lint openapi.yml
```

Either you use `$ spectral lint openapi.yml` if it is installed in your $PATH, or you run the command from its downloaded location: `$ ~/Downloads/spectral-macos lint openapi.yml` but what you are doing is not expected to work. 

LMK if this fixes it!

**Comment by nogates** [comment]
Hey @philsturgeon! Thanks for the quick reply :)

 > I am a little confused by the command you are running here:

Sorry, I should have removed that part for clarity, but the first `spectral` is jut my `zsh` prompt (it just tells me what folder I am in)

```
-> spectral pwd
/tmp/spectral
```

Just for clarity, using the script and downloading the binary to the default location, produces the same error:

```
~ which spectral
/usr/local/bin/spectral
~ spectral lint openapi.yml
Error: @stoplight/path: Cannot parse /Users/ddujo/openapi.yml because it is not a string
    at parse (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:6984)
    at Object.isAbsolute (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:7876)
    at Object.<anonymous> (/snapshot/spectral/dist/cli/services/linter.js:40:65)
    at Generator.next (<anonymous>)
    at /snapshot/spectral/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/snapshot/spectral/node_modules/tslib/tslib.js:106:16)
    at Object.lint (/snapshot/spectral/dist/cli/services/linter.js:36:20)
    at Object.handler (/snapshot/spectral/dist/cli/commands/lint.js:86:25)
    at Object.runCommand (/snapshot/spectral/node_modules/yargs/lib/command.js:235:26)
➜  ~
```

**Comment by philsturgeon** [comment]
Thanks for clearing that up nogates, makes a lot more sense :D 

What is the contents of openapi.yml?

**Comment by nogates** [comment]
> What is the contents of openapi.yml?

Doesn't really matter. All the oas yml files I've tried fail with the same error. For instance, downloading the default petstore from the oas repo also fails

```
➜  ~ curl https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v3.0/petstore.yaml --output petstore.yaml
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2573  100  2573    0     0   3387      0 --:--:-- --:--:-- --:--:--  3385
➜  ~ spectral lint petstore.yaml
Error: @stoplight/path: Cannot parse petstore.yaml because it is not a string
    at parse (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:6984)
    at Object.isAbsolute (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:7876)
    at Object.<anonymous> (/snapshot/spectral/dist/cli/services/linter.js:40:65)
    at Generator.next (<anonymous>)
    at /snapshot/spectral/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/snapshot/spectral/node_modules/tslib/tslib.js:106:16)
    at Object.lint (/snapshot/spectral/dist/cli/services/linter.js:36:20)
    at Object.handler (/snapshot/spectral/dist/cli/commands/lint.js:86:25)
    at Object.runCommand (/snapshot/spectral/node_modules/yargs/lib/command.js:235:26)
```

and as I said, the same file with the old 4.1.1 version works as expected

```
➜  ~ Downloads/spectral-macos lint petstore.yaml
OpenAPI 3.x detected

/Users/ddujo/petstore.yaml
  2:6   warning  info-contact           Info object should contain `contact` object.
  2:6   warning  info-description       OpenAPI object info `description` must be present and non-empty string.
  11:9  warning  operation-description  Operation `description` must be present and non-empty string.
 42:10  warning  operation-description  Operation `description` must be present and non-empty string.
  57:9  warning  operation-description  Operation `description` must be present and non-empty string.

✖ 5 problems (0 errors, 5 warnings, 0 infos)
```

**Comment by philsturgeon** [comment]
Confirmed! 

```
$ /usr/local/bin/spectral
spectral <command>

Commands:
  spectral lint <documents..>  lint JSON/YAML documents from files or URLs

Options:
  --version  Show version number                                                            [boolean]
  --help     Show help                                                                      [boolean]

$ /usr/local/bin/spectral --version
0.0.0

$ /usr/local/bin/spectral lint ../prism/examples/petstore.oas2.json
Error: @stoplight/path: Cannot parse ../prism/examples/petstore.oas2.json because it is not a string
    at parse (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:6984)
    at Object.isAbsolute (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:7876)
    at Object.<anonymous> (/snapshot/spectral/dist/cli/services/linter.js:40:65)
    at Generator.next (<anonymous>)
    at /snapshot/spectral/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/snapshot/spectral/node_modules/tslib/tslib.js:106:16)
    at Object.lint (/snapshot/spectral/dist/cli/services/linter.js:36:20)
    at Object.handler (/snapshot/spectral/dist/cli/commands/lint.js:86:25)
    at Object.runCommand (/snapshot/spectral/node_modules/yargs/lib/command.js:235:26)
```

While we fix this up, have you had any luck with docker or NPM? See the [installation instructions](https://stoplight.io/p/docs/gh/stoplightio/spectral/docs/getting-started/installation.md) for the them.

**Comment by philsturgeon** [comment]
With the help of @P0lip I think we got this fixed. Thank you so much for noticing, head over to #792 for the fix.

**Comment by P0lip** [comment]
@nogates @philsturgeon 
I updated updated spectral-macos 4.2.0 binary.
LMK if it works for you all, please.

**Comment by nogates** [comment]
It's working now! thank @P0lip and @philsturgeon !
